tools/mlir-gen/MLIRGen.cpp

```diff
@@ -139,15 +169,30 @@ Value MLIRGenerator::createLayer(LayerArgs &args) {
   OpBuilder::InsertionGuard guard(builder);
 
   Value chain;
-  chain = lowerMatmul(args.input.value, args.weight.value, args.output.value);
+  if (outputOpKind == OutputOpKind::Generic) {
```

Comment:
Since `outputKind` is a field of the class, the `lowerMatmul` function can just use it directly, so the if can be inside the `lower*` functions, not here.